crates/sui-graphql-rpc/src/types/query.rs

```diff
@@ -48,13 +49,16 @@ pub(crate) type SuiGraphQLSchema = async_graphql::Schema<Query, Mutation, EmptyS
 
 #[Object]
 impl Query {
-    /// First four bytes of the network's genesis checkpoint digest (uniquely identifies the
-    /// network).
     async fn chain_identifier(&self, ctx: &Context<'_>) -> Result<String> {
-        Ok(ChainIdentifier::query(ctx.data_unchecked())
-            .await
-            .extend()?
-            .to_string())
+        let chain_id: ChainIdentifier = *ctx.data()?;
```

Comment:
Let's use `data_unchecked` here -- it will cause the system to panic straight away if the value has not been set. Most of the time we want to avoid that, but in this case it is better to panic because if this value does not exist, it means the service has been misconfigured, and we'd rather find that out early than wait for a report from a user about a confusing error message.